### PR TITLE
Add start/stop/restart instance

### PR DESCRIPTION
**Pull request by pintea-eugen**

### CODE REVIEW

File: cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstancesRestart.scala [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  * Created by pinteae on 3/2/2017.
+  */
+class InstancesRestart {
+
+  @Action(name = "Restart Instance",
+    outputs = Array(
+      new Output(OutputNames.RETURN_CODE),
+      new Output(OutputNames.RETURN_RESULT),
+      new Output(OutputNames.EXCEPTION)
+    ),
```
Comment (robert-sandor) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstancesRestart.scala]: i think you forgot to expose the ZONE_OPERATION_NAME output

File: cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstancesStart.scala [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+/**
+  * Created by pinteae on 3/1/2017.
+  */
+class InstancesStart {
+  @Action(name = "Start Instance",
+    outputs = Array(
+      new Output(OutputNames.RETURN_CODE),
+      new Output(OutputNames.RETURN_RESULT),
+      new Output(OutputNames.EXCEPTION)
+    ),
```
Comment (robert-sandor) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstancesStart.scala]: same as before

File: cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstancesStop.scala [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+/**
+  * Created by pinteae on 3/2/2017.
+  */
+class InstancesStop {
+  @Action(name = "Stop Instance",
+    outputs = Array(
+      new Output(OutputNames.RETURN_CODE),
+      new Output(OutputNames.RETURN_RESULT),
+      new Output(OutputNames.EXCEPTION)
+    ),
```
Comment (robert-sandor) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstancesStop.scala]: same as before

File: cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/services/compute/instances/InstanceService.scala [hunk 4]
```diff
@@ -35,6 +35,20 @@ object InstanceService {
...
+      .execute()
+
+  def stop(httpTransport: HttpTransport, jsonFactory: JsonFactory, credential: Credential, project: String, zone: String, instanceName: String): Operation =
+    ComputeService.instancesService(httpTransport, jsonFactory, credential)
+      .stop(project, zone, instanceName)
+      .execute()
+
+  def restart(httpTransport: HttpTransport, jsonFactory: JsonFactory, credential: Credential, project: String, zone: String, instanceName: String): Operation =
+    ComputeService.instancesService(httpTransport, jsonFactory, credential)
+      .reset(project, zone, instanceName)
```
Comment (victorursan) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/services/compute/instances/InstanceService.scala]: check if this is what we need and not what we want :laughing:

File: cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstancesStop.scala [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+              @Param(value = ZONE, required = true) zone: String,
+              @Param(value = INSTANCE_NAME, required = true) instanceName: String,
+              @Param(value = ACCESS_TOKEN, required = true, encrypted = true) accessToken: String,
+              @Param(value = PROXY_HOST) proxyHost: String,
+              @Param(value = PROXY_PORT) proxyPortInp: String,
+              @Param(value = PROXY_USERNAME) proxyUsername: String,
+              @Param(value = PROXY_PASSWORD, encrypted = true) proxyPasswordInp: String,
+              @Param(value = PRETTY_PRINT) prettyPrintInp: String): util.Map[String, String] = {
+
+    val proxyHostOpt = InputUtils.verifyEmpty(proxyHost)
```
Comment (victorursan) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstancesStop.scala]: import verifyEmpty

File: cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstancesRestart.scala [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+  def execute(@Param(value = PROJECT_ID, required = true) projectId: String,
+              @Param(value = ZONE, required = true) zone: String,
+              @Param(value = INSTANCE_NAME, required = true) instanceName: String,
+              @Param(value = ACCESS_TOKEN, required = true) accessToken: String,
+              @Param(value = PROXY_HOST) proxyHost: String,
+              @Param(value = PROXY_PORT) proxyPortInp: String,
+              @Param(value = PROXY_USERNAME) proxyUsername: String,
+              @Param(value = PROXY_PASSWORD, encrypted = true) proxyPasswordInp: String,
+              @Param(value = PRETTY_PRINT) prettyPrintInp: String): util.Map[String, String] = {
+    val proxyHostOpt = InputUtils.verifyEmpty(proxyHost)
```
Comment (victorursan) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstancesRestart.scala]: import verifyEmpty

File: cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstancesRestart.scala [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+      return getFailureResultsMap(validationStream.mkString(NEW_LINE))
+    }
+
+    val proxyPort = toInteger(proxyPortStr)
+    val prettyPrint = toBoolean(prettyPrintStr)
+
+    try {
+      val httpTransport = HttpTransportUtils.getNetHttpTransport(proxyHostOpt, proxyPort, proxyUsernameOpt, proxyPasswordStr)
+      val jsonFactory = JsonFactoryUtils.getDefaultJacksonFactory
+
```
Comment (victorursan) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstancesRestart.scala]: too many new lines ( group some of them )

File: cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/services/compute/instances/InstanceService.scala (hunk 4 above)
Comment (pintea-eugen) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/services/compute/instances/InstanceService.scala]: Yes, this is :)

Comment (victorursan) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/services/compute/instances/InstanceService.scala]: so it waits for 2 minutes before it hard resets it ...

**Pull request merged**